### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by gusty**
This will add the infrastructure to build docs, together with some initial documentation.

### CODE REVIEW

**Comment by gusty** [comment]
I just released the docs using this PR, they are here: https://jet.github.io/XRay/

Anyway the idea of this PR is mainly the docs infrastructure, docs can be changed/added later on a separate PR.

File: buildDocs.fsx [hunk 1]
```diff
@@ -0,0 +1,1456 @@
+// Generate and Release Docs
```
Comment (vchekan) [inline buildDocs.fsx:1]: 1456 lines of code for build script beg for more explanation then 1 sentence in commit message. It feels like inlined FAKE.

Comment (gusty) [reply buildDocs.fsx:1]: The idea is to squash this as a single commit, so we can capture what it takes to add doc infrastructure to a project, but I'm open to other suggestion.

It contains many inlined portions of FAKE as it states in the following lines.

This way, we can build and release the docs with a single command line `fsi buildDocs.fsx ReleaseDocs`

Comment (gusty) [reply buildDocs.fsx:1]: I've splitted the file in two, it's more readable now.

**Pull request merged**